src/components/NotePage/NoteDetail/NoteDetail.tsx

```diff
@@ -299,7 +299,8 @@ export default class NoteDetail extends React.Component<
   }
 
   appendNewTag = () => {
-    if (isTagNameValid(this.state.newTagName)) {
+    const includes = this.state.tags.includes(this.state.newTagName)
+    if (isTagNameValid(this.state.newTagName) && !includes) {
```

Comment:
Can you fix it like the below? 
1. Discard nested structure by excluding invalid cases at the first.
2. Use `includes` function from ramda. (It looks better for me because `this.state.tags.includes` is too long to read :eyes:)

```ts
import { includes } from 'ramda'

...

if (includes(this.state.newTagName, this.state.tags)) { return }
if (isTagNameValid(this.state.newTagName)) { return }
this.setState(...)
```

I'll fix it by myself if this change request is not resolved in 6 hours.

Reply:
Thanks to review😄

>(It looks better for me because this.state.tags.includes is too long to read 👀

I agree with your opinion:+1:

I fixed this commit: 7de2708